lib/_http_outgoing.js

```diff
@@ -728,8 +728,23 @@ OutgoingMessage.prototype.setHeaders = function setHeaders(headers) {
     throw new ERR_INVALID_ARG_TYPE('headers', ['Headers', 'Map'], headers);
   }
 
-  for (const key of headers.keys()) {
-    this.setHeader(key, headers.get(key));
+  // Headers object joins multiple cookies with a comma when using
+  // the getter to retrieve the value,
+  // unless iterating over the headers directly.
+  // We also cannot safely split by comma.
+  // To avoid setHeader owerwriting the previous value we push
+  // set-cookie values in array and set them all at once.
+  const cookies = [];
+
+  for (const { 0: key, 1: value } of headers.entries()) {
+    if (key === 'set-cookie') {
+      cookies.push(value);
```

Comment:
The value can be an `array`, directly push here is not recommended.

Reply:
What do you think would be more appropriate?

Reply:
`ArrayIsArray(value) ? ...value : value`

Reply:
Oh true, I wasnt considering map 🌝 fixed and added a test